android/src/com/frostwire/android/gui/adapters/menu/RenameFileMenuAction.java

```diff
@@ -133,12 +145,14 @@ protected void initComponents(Dialog dlg, Bundle savedInstanceState) {
             noButton.setOnClickListener(new ButtonOnClickListener(this, false));
         }
 
-        public void rename() {
+        public String rename() {
             EditText input = findView(getDialog(), R.id.dialog_default_input_text);
+            String newFileName = new String(input.getText().toString());
```

Comment:
I made a copy to make sure the returned value wouldn't be a weird reference (mem leak or lost reference) to the dialog that's about to be dismissed.